tests/baselines/reference/arrowFunctionTypeWithUnion.types

```diff
@@ -0,0 +1,12 @@
+=== tests/cases/compiler/arrowFunctionTypeWithUnion.ts ===
+declare const a: string | (a: string, b: boolean) => string
+>a : string | ((a: string, b: boolean) => string)
+>a : string
+>b : boolean
+
+declare const b: string | () => string
+>b : string | (() => string)
+
+declare const c: string | () => string | number | () => boolean
+>c : string | (() => string | number | (() => boolean))

```

Comment:
How can you tell `() => boolean` is not part of the return type of the preceding function type?

Reply:
obviously  parens😂
that is currently parse rules, this pr only allow us write less parens